spec/app/ics-027-interchain-accounts/README.md

```diff
@@ -20,110 +20,126 @@ On Ethereum, there are two types of accounts: externally owned accounts, control
 
 ### Definitions
 
+**TODO: Make this a table and add more useful definitions for used keywords throughout the spec**
+
 The IBC handler interface & IBC relayer module interface are as defined in [ICS 25](../ics-025-handler-interface) and [ICS 26](../ics-026-routing-module), respectively.
 
 ### Desired Properties
 
 - Permissionless
-- Fault containment: Interchain account must follow rules of its host chain, even in times of Byzantine behaviour by the counterparty chain (the chain that manages the account)
-- The chain that controls the account must process the results asynchronously and according to the chain's logic. The result's code should be 0 if the transaction was successful and an error code other than 0 if the transaction failed.
-- Sending and receiving transactions will be processed in an ordered channel where packets are delivered exactly in the order which they were sent.
+- Fault containment: An interchain account must follow rules of its host chain, even in times of Byzantine behavior by the counterparty chain (the chain that manages the account)
+- The chain that controls the account must process the results asynchronously and according to the chain's logic. The acknowledgment message should contain a result of an error as described in [ics-4](https://github.com/cosmos/ibc/tree/master/spec/core/ics-004-channel-and-packet-semantics#acknowledgement-envelope).
```

Comment:
I don't understand `according to the chain's logic`. This seems to be referring to receiving acknowledgements but what logic is done when processing a failed/successful acknowledgement?

Or maybe the wording is incorrect? Do you think it is referring to the OnRecv or OnAcknowledge callbacks?

Reply:
Good point, I'll update this.

Reply:
I don't see a problem with this beyond that it's maybe irrelevant/distracting information. The origin chain may have custom logic like: once first tx is submitted and successfully processed create a new one and begin the process of submitting it. Basically the origin chain is must process the results async and is _able_ to use that ack within it's own logic.

Reply:
```suggestion
- The chain that controls the account must process the results asynchronously. The acknowledgment message should contain a result of an error as described in [ics-4](https://github.com/cosmos/ibc/tree/master/spec/core/ics-004-channel-and-packet-semantics#acknowledgement-envelope) which can be used as part of the chain's logic.
```

Reply:
> result of an error

"result *or* error..."?